nilearn/_utils/data_gen.py

```diff
@@ -797,8 +797,9 @@ def create_fake_bids_dataset(base_dir='',
                              with_confounds=True,
                              confounds_tag="desc-confounds_timeseries",
                              no_session=False,
-                             random_state=0):
-    """Creates a fake :term:`bids<BIDS>` dataset directory with dummy files.
+                             random_state=0,
+                             entities=None,):
```

Comment:
This is currently a very limited way to create more diverse filenames: can only add one entity associated with a list of potential labels.

for example: `['ce', ['A', 'B']]`

This should be fine when mocking common BIDS datasets but may fall short for more complex edge cases.

Reply:
May be better to use a dict instead of a list.

Reply:
The question is whether the keys need to be picked within a fixed vocabulary.

Reply:
Can at least allow all the typical entities from BIDS for MRI data (raw and derivatives)

Reply:
There no such thing as an exhaustive list of those ?

Reply:
List of all BIDS entities: 
https://bids-specification.readthedocs.io/en/latest/appendices/entities.html

To specifically know what entities are allowed for functional data (for example), you then
must turn to the BIDS schema (and speak YAML fluently):
- raw func:  https://github.com/bids-standard/bids-specification/blob/9e607f8bab02c546d6ad0ca9e7995d9ca02db905/src/schema/rules/files/raw/func.yaml#L2
- raw derivatives: https://github.com/bids-standard/bids-specification/blob/master/src/schema/rules/files/deriv/imaging.yaml#L74